docs/source/reference/types.rst

```diff
@@ -0,0 +1,163 @@
+Types, Functions, and Interfaces
+================================
+
+The type system in Idris 2 is quite powerful, and is the main feature that sets it apart from simply typed languages like Haskell or OCaml.
+
+There are many ways to construct types, those being members of ``Type``, in Idris.
+
+-------------
+Functions
+-------------
+
+Function types, formed by ``(x : A) -> B``, are valid types, given that ``A`` is a valid type *or* kind and ``B`` is a valid type *or* kind. 
+As shorthand, for non-dependent types, ``A -> B`` is shorthand for ``(_ : A) -> B``.
+
+^^^^^^^^^^^^^
+Implicits
+^^^^^^^^^^^^^
+
+Implicit function types, those where the value of a parameter can be inferred from the context, can take three forms.
+The first of these, ``{x : A} -> B``, forms the basis "basic" implicit types. 
+That is, ``x`` should be inferred *syntactically*. 
+This is the equivalent of implicit ``forall`` in Haskell or Rocq, where we might state that ``vectorMap`` on a vector might be
+
+.. code:: idris
+
+    vectorMap : {n : Nat} -> (a -> b) -> (Vect n a) -> (Vect n b)
+
+We would then use ``vectorMap`` as follows: 
+
+.. code-block:: idris
+
+    vectorMap f v
+
+or
+
+.. code:: idris
+
+    vectorMap {n = m} f v
+
+These two examples show the power of Idris implicits: while they can be inferred by the compiler, if necessary (or desired) they may be specified by the user
+
+.. tip:: 
+    Idris *not have* a ``forall`` keyword, unlike Haskell. 
```

Comment:
Idris does have a `forall`, but I don't know if it differs in meaning from that in Haskell

Reply:
`forall x . <stuff>` is `{0 x : _} -> <stuff>`